rskj-core/src/main/java/co/rsk/peg/pegininstructions/PeginInstructionsProvider.java

```diff
@@ -0,0 +1,37 @@
+package co.rsk.peg.pegininstructions;
+
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.peg.utils.BtcTransactionFormatUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PeginInstructionsProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(PeginInstructionsProvider.class);
+
+    public PeginInstructionsBase buildPeginInstructions(BtcTransaction btcTx) throws Exception {
```

Comment:
```suggestion
    public PeginInstructions buildPeginInstructions(BtcTransaction btcTx) throws PeginInstructionsException {
```
You should return the interface, not the base class, and the method should throw a specific Exception, not a generic one